src/gwt/src/org/rstudio/studio/client/workbench/views/environment/model/MemoryUsage.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * MemoryStat.java
+ *
+ * Copyright (C) 2021 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.studio.client.workbench.views.environment.model;
+
+import com.google.gwt.core.client.JavaScriptObject;
+
+public class MemoryUsage extends JavaScriptObject
+{
+   protected MemoryUsage() {}
+
+   public final native MemoryStat getTotal() /*-{
+      return this.total;
+   }-*/;
+
+   public final native MemoryStat getUsed() /*-{
+      return this.used;
+   }-*/;
+
+   public final native MemoryStat getProcess() /*-{
+      return this.process;
+   }-*/;
+
+   /**
+    * Compute the percentage of memory used.
+    *
+    * @return The amount of memory used as a percentage of the total.
+    */
+   public final int getPercentUsed()
+   {
+      return (int)Math.round(((getUsed().getKb() * 1.0) / (getTotal().getKb() * 1.0)) * 100);
+   }
+
+   /**
+    * Gets a color code representing the memory usage (green = low, red = high, etc.)
+    *
+    * @return The color code, in a CSS-compatible format.
+    */
+   public final String getColorCode()
```

Comment:
Showing information entirely via color is a general no-no for both color blindness issues and for screen reader users. I don't have any proposals for quick fixes, and consider this is fine as-is (the entire environment pane has significant issues for screen reader users and this would be WAY down the list of things to worry about!).

Reply:
The color is just a visual decoration for the percentage value, which is accessible in several other ways (both visual and text), so I don't think this is a concern as implemented.